### ISSUE TITLE
Review patches for #162, #160 and wrong data_type return at oracle_statement_backend::describe_column

### ISSUE THREAD

**Issue by mloskot** [SOCI/soci#197]
The original PR #168 has not been merged, but automatically closed as result of deleting `hotfix/3.2.2` branch after releasing SOCI 3.2.2.

The PR attempts to fix #162, #160 and wrong data_type return at oracle_statement_backend::describe_column.

The PR is worth to review and may still be merged into future releases, so let's keep this issue open as reminder.

**Comment by Alexander-Shukaev** [comment]
Hello.  Just out of curiosity, is the project abandoned?  I don't see any active development going on neither on `master` nor on any other branch.  At the same time, this contribution is more than 3 years old and is still unreviewed.  Is the **4.0.0** release still on track or there are some other things in the way?  It looks like this milestone includes a lot of work, the progress of which is not easily visible, at least though this repository.

**Comment by mloskot** [comment]
The maintenance and development pace has slowed down indeed, but I don't think it is abandoned. Personally, I'm unable to keep it up to pursue the 4.0.0 goal and I wouldn't be able to support it after releasing.

Most of the recent improvements come from the library users. Hence, the development cycle switched to, sort of, a rolling release with the current master recommended to use.

**Comment by vadz** [comment]
Indeed, it's not abandoned in the sense that we, for example, continue to actively use it and I will try to at least take a look at any PRs that might be affecting us -- but we don't use Oracle, for example, and I don't know much about it, so I can't really review this particular one.

Any testing/feedback from Oracle users would be welcome.

**Comment by mloskot** [comment]
I can also confirm that I try to review any PRs and merge any fixes and improvements.

If we agree on feasible roadmap of fixes and features, PRs to merge and Issues to close, for SOCI 4.0.0, then we can try to work towards completing the goal.

For me personally, I've been hopeful about two items from the [Roadmap](https://github.com/SOCI/soci/wiki/Roadmap) I compiled a while ago:

1. [Integer conversions overhaul](https://github.com/SOCI/soci/wiki/RFC1) which I haven't even started. Since it might be compatibility breaker, I didn't want to release 4.0.0 without completing this task.

2. New documentation (#332) - content has been migrated to Markdown already. Next, I need to set up Sphinx generator, some infrastructure update (eg. with GitHub pages) and update docs wherever needed to catch up with the changes in SOCI 4.

Everyone is welcome to help, absolutely.
